test/integration/acceptance/add_signatory_test.cpp

```diff
@@ -0,0 +1,172 @@
+      .sendTxAwait(
+          makeFirstUser({interface::permissions::Role::kAddMySignatory}),
+          CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTxAwait(
+          complete(baseTx().grantPermission(
+              kUser2Id, interface::permissions::Grantable::kAddMySignatory)),
+          CHECK_BLOCK(1))
+      .sendTxAwait(complete(baseTx().creatorAccountId(kUser2Id).addSignatory(
+                                kUserId, kUser2Keypair.publicKey()),
+                            kUser2Keypair),
+                   CHECK_BLOCK(1));
+}
+
+/**
+ * C226 Add signatory to account, which haven't granted such permission
+ * @given some user with CanAddMySignatory permission and a second user without
+          granted CanAddMySignatory
+ * @when execute tx with AddSignatory where the second is a creator and the
+ *       second's key is added as a signatory
+ * @then there is no tx in proposal
+ */
+TEST_F(AddSignatory, NonGrantedPermission) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(
+          makeFirstUser({interface::permissions::Role::kAddMySignatory}),
+          CHECK_BLOCK(1))
+      .sendTxAwait(makeSecondUser(), CHECK_BLOCK(1))
+      .sendTx(complete(baseTx().creatorAccountId(kUser2Id).addSignatory(
+                           kUserId, kUser2Keypair.publicKey()),
+                       kUser2Keypair))
+      .checkVerifiedProposal(CHECK_BLOCK(0));
+}
+
+/**
+ * C222 Add signatory to non-existing account ID
+ * @given some user with CanAddMySignatory permission
+ * @when execute tx with AddSignatory with inexistent user
+ * @then there is no tx in proposal
+ */
+TEST_F(AddSignatory, NonExistentUser) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(makeFirstUser(), CHECK_BLOCK(1))
+      .sendTx(complete(baseTx().addSignatory("inexistent@" + kDomain,
+                                             kUserKeypair.publicKey()),
+                       kUser2Keypair))
+      .checkVerifiedProposal(CHECK_BLOCK(0));
+}
+
+/**
+ * C223 Add invalid public key
+ * @given some user with CanAddMySignatory permission
+ * @when execute tx with AddSignatory with inexistent public key
+ * @then the tx is stateless invalid
+ */
+TEST_F(AddSignatory, InvalidKey) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(makeFirstUser(), CHECK_BLOCK(1))
+      .sendTx(complete(baseTx().addSignatory(kUserId,
+                                             shared_model::crypto::PublicKey(
+                                                 std::string(1337, 'a'))),
+                       kUser2Keypair),
+              checkStatelessInvalid);
+}
+
+/**
+ * @given some user with CanAddMySignatory permission
+ * @when execute tx with AddSignatory with a valid key which isn't associated
+ *       with any user
+ * @then there is no tx in proposal
+ */
+TEST_F(AddSignatory, NonexistedKey) {
```

Comment:
Maybe use "NonExistent" for correctness and to be similar to NonExistentUser test